olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -168,7 +184,13 @@ bool DefaultCacheImpl::Put(const std::string& key, const boost::any& value,
   auto encoded_item = encoder();
   if (memory_cache_) {
     const auto size = encoded_item.size();
-    const bool result = memory_cache_->Put(key, value, expiry, size);
+    // reset expiry if key is protected only for memory cache, in mutable cache
```

Comment:
If a tile is protected it should not have any expiry also. Please also reset the expiry for the mutable cache.
And if a protected cache entry cannot expire you should not need to have all this "reset expiry" blocks as the expiry value in the mutable cache is already kDefaultExpiry.